src/app/tests/TestAclEvent.cpp

```diff
@@ -0,0 +1,372 @@
+
+    ChipLogDetail(DataManagement, "Generating Events");
+    testEventGenerator.SetStatus(0);
+    err = logMgmt.LogEvent(&testEventGenerator, options1, eid1);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    testEventGenerator.SetStatus(1);
+    err = logMgmt.LogEvent(&testEventGenerator, options2, eid2);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+}
+
+class MockInteractionModelApp : public chip::app::ReadClient::Callback
+{
+public:
+    void OnEventData(const chip::app::EventHeader & aEventHeader, chip::TLV::TLVReader * apData,
+                     const chip::app::StatusIB * apStatus) override
+    {
+        ++mNumDataElementIndex;
+        mGotEventResponse = true;
+        if (apStatus != nullptr && !apStatus->IsSuccess())
+        {
+            mNumReadEventFailureStatusReceived++;
+            mLastStatusReceived = *apStatus;
+        }
+        else
+        {
+            mLastStatusReceived = chip::app::StatusIB();
+        }
+    }
+
+    void OnError(CHIP_ERROR aError) override
+    {
+        mError     = aError;
+        mReadError = true;
+    }
+
+    void OnDone(chip::app::ReadClient *) override {}
+
+    void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override
+    {
+        if (aReadPrepareParams.mpEventPathParamsList != nullptr)
+        {
+            delete[] aReadPrepareParams.mpEventPathParamsList;
+        }
+    }
+
+    int mNumDataElementIndex               = 0;
+    bool mGotEventResponse                 = false;
+    int mNumReadEventFailureStatusReceived = 0;
+    int mNumAttributeResponse              = 0;
+    bool mGotReport                        = false;
+    bool mReadError                        = false;
+    chip::app::ReadHandler * mpReadHandler = nullptr;
+    chip::app::StatusIB mLastStatusReceived;
+    CHIP_ERROR mError = CHIP_NO_ERROR;
+    std::vector<chip::app::ConcreteAttributePath> mReceivedAttributePaths;
+};
+
+} // namespace
+
+namespace chip {
+namespace app {
+class TestAclEvent
+{
+public:
+    static void TestReadRoundtripWithEventStatusIBInEventReport(nlTestSuite * apSuite, void * apContext);
+};
+
+void TestAclEvent::TestReadRoundtripWithEventStatusIBInEventReport(nlTestSuite * apSuite, void * apContext)
+{
+    TestContext & ctx = *static_cast<TestContext *>(apContext);
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+
+    Messaging::ReliableMessageMgr * rm = ctx.GetExchangeManager().GetReliableMessageMgr();
+    // Shouldn't have anything in the retransmit table when starting the test.
+    NL_TEST_ASSERT(apSuite, rm->TestGetCountRetransTable() == 0);
+
+    GenerateEvents(apSuite, apContext);
+
+    auto * engine = chip::app::InteractionModelEngine::GetInstance();
+    err           = engine->Init(&ctx.GetExchangeManager(), &ctx.GetFabricTable());
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    // When reading events with concrete paths without enough privilege, we will get a EventStatusIB
```

Comment:
Please add comments here above the test to indicate that a custom AccessControl::Delegate has been installed that grants privilege to any cluster except the test cluster.